hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -114,7 +115,17 @@ int QuectelNcpClient::init(const NcpClientConfig& conf) {
     conf_ = static_cast<const CellularNcpClientConfig&>(conf);
 
     // Initialize serial stream
-    auto sconf = SERIAL_8N1 | SERIAL_FLOW_CONTROL_NONE; // TODO: replace with SERIAL_FLOW_CONTROL_RTS_CTS when HW is fixed.
+    auto sconf = SERIAL_8N1 | SERIAL_FLOW_CONTROL_RTS_CTS;
+
+    // Hardware version
+    // V003 - 0x00 (disable hwfc)
+    // V004 - 0x01 (enable hwfc)
+    uint32_t hwVersion = 0xFF;
+    hal_get_device_hw_version(&hwVersion, nullptr);
+    if (hwVersion == 0x00) {
```

Comment:
Let's make sure we check the return code from `hal_get_device_hw_version` as well. In `initReady()` as well.

Reply:
Fixed